testsuite/standalone/src/test/java/org/wildfly/core/test/standalone/mgmt/api/core/DeploymentOperationsTestCase.java

```diff
@@ -0,0 +1,415 @@
+import java.io.File;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Properties;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import javax.inject.Inject;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.OperationBuilder;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.controller.client.helpers.standalone.DeploymentAction;
+import org.jboss.as.controller.client.helpers.standalone.DeploymentPlan;
+import org.jboss.as.controller.client.helpers.standalone.ServerDeploymentActionResult;
+import org.jboss.as.controller.client.helpers.standalone.ServerDeploymentManager;
+import org.jboss.as.controller.client.helpers.standalone.ServerDeploymentPlanResult;
+import org.jboss.as.test.deployment.trivial.ServiceActivatorDeploymentUtil;
+import org.jboss.as.test.shared.TimeoutUtil;
+import org.jboss.dmr.ModelNode;
+import org.jboss.shrinkwrap.api.exporter.ExplodedExporter;
+import org.jboss.shrinkwrap.api.exporter.ZipExporter;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.core.testrunner.ManagementClient;
+import org.wildfly.core.testrunner.WildflyTestRunner;
+
+/**
+ * Tests to check hard requirements and extreme cases for deployment operations.
+ * @author <a href="mailto:mjurc@redhat.com">Michal Jurc</a> (c) 2016 Red Hat, Inc.
+ */
+@RunWith(WildflyTestRunner.class)
+public class DeploymentOperationsTestCase {
+
+    @Inject
+    private ManagementClient managementClient;
+
+    private static final int TIMEOUT = TimeoutUtil.adjust(20000);
+    private static final String TEST_DEPLOYMENT_NAME = "test-deployment.jar";
+
+    @After
+    public void cleanDeployments() {
+        ModelNode op = new ModelNode();
+        op.get(OP).set(READ_RESOURCE_OPERATION);
+        op.get(OP_ADDR).set(DEPLOYMENT, TEST_DEPLOYMENT_NAME);
+        ModelNode result = awaitOperationExecutionAndReturnOutcome(op);
+        if (Operations.isSuccessfulOutcome(result)) {
+            undeployAndRemoveDeployment(TEST_DEPLOYMENT_NAME);
+        }
+    }
+
+    @AfterClass
+    public static void cleanFiles() {
+        String jbossBaseDir = System.getProperty("jboss.home");
+        Assert.assertNotNull(jbossBaseDir);
+        Path dataDir = new File(jbossBaseDir).toPath().resolve("standalone").resolve("data");
+        Assert.assertTrue(Files.exists(dataDir));
+        cleanFile(dataDir.resolve("managed-exploded").toFile());
+        cleanFile(new File("target/archives/" + TEST_DEPLOYMENT_NAME));
```

Comment:
You should delete the archives directory which you have created :so this should be cleanFile(new File("target", "archives"));